### PR TITLE
Add Vault List

### PR DESCRIPTION
**Pull request by kalashshah**
### Ticket Number

- [1684](https://github.com/push-protocol/push-dapp/issues/1684)

### Description

- **Problem/Feature**:
- Implements the UI and functionality for the vault list

### Type of Change

- [x] New feature

### Checklist

- [ ] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [ ] Quick PR label added
- [ ] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [ ] DO NOT Merge PR label added

### Frontend Guidelines

- [x] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [ ] Engineer has tested the changes on deploy preview

### Review & Approvals

- [x] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### CODE REVIEW

File: src/queries/hooks/pointsVault/useGetPointsVaultUsers.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  pointsVaultApprovedUsers,
+  pointsVaultPendingUsers,
+  pointsVaultRejectedUsers,
+} from '../../queryKeys';
+import { PointsVaultGetUsersPayload } from 'queries/types';
+import { getPointsVaultUsers } from 'queries/services';
+import { usePointsVaultToken } from './usePointsVaultToken';
+
+export const useGetPointsVaultUsers = ({ status, pageSize = 20 }: PointsVaultGetUsersPayload) => {
+  const token = usePointsVaultToken();
```
Comment (rohitmalhotra1420) [inline src/queries/hooks/pointsVault/useGetPointsVaultUsers.ts:13]: This needs to be used in the component.

File: src/queries/hooks/pointsVault/useApprovePVUserMutation.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ? [pointsVaultUsers, pointsVaultRejectedUsers]
+    : [pointsVaultUsers, pointsVaultPendingUsers];
+};
+
+type UseApprovePVUserProps = {
+  onSuccess?: () => void;
+  onError?: () => void;
+};
+
+export const useApprovePVUser = ({ onSuccess, onError }: UseApprovePVUserProps) => {
```
Comment (rohitmalhotra1420) [inline src/queries/hooks/pointsVault/useApprovePVUserMutation.ts:23]:
```suggestion
export const useApprovePointsVaultUser = ({ onSuccess, onError }: UseApprovePVUserProps) => {
```

File: src/queries/hooks/pointsVault/useRejectPVUserMutation.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    ? [pointsVaultUsers, pointsVaultApprovedUsers]
+    : [pointsVaultUsers, pointsVaultPendingUsers];
+};
+
+type UseRejectPVUserProps = {
+  onSuccess?: () => void;
+  onError?: () => void;
+};
+
+export const useRejectPVUser = ({ onError, onSuccess }: UseRejectPVUserProps) => {
```
Comment (rohitmalhotra1420) [inline src/queries/hooks/pointsVault/useRejectPVUserMutation.ts:23]:
```suggestion
export const useRejectPointsVaultUser = ({ onError, onSuccess }: UseRejectPVUserProps) => {
```

File: src/queries/hooks/pointsVault/useApprovePVUserMutation.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    onSettled: async (_, __, { currentStatus }) => {
+      const keys = getKeysFromStatus(currentStatus);
+      await queryClient.invalidateQueries({ queryKey: keys });
+      await queryClient.invalidateQueries({ queryKey: [pointsVaultUsers, pointsVaultApprovedUsers] });
+      onSuccess?.();
+    },
+  });
+
+  return { approvePVUserMutation: mutation };
```
Comment (rohitmalhotra1420) [inline src/queries/hooks/pointsVault/useApprovePVUserMutation.ts:51]:
```suggestion
  => useMutation( { mutationFn: approvePointsVaultUser , mutationKey: approveVaultUserKey} )
```

Comment (rohitmalhotra1420) [reply src/queries/hooks/pointsVault/useApprovePVUserMutation.ts:51]: We will use onSuccess and onError callback function provided by mutation function.

```

// payload is the data passed to network call function

mutation(payload, {

onSuccess:(response)=>{
refetchApprovedUserList()
},
 onError:(response)=>{
if(response.error===403){ // token is expired
 setView("login")
}
}
} )

```

File: src/queries/hooks/pointsVault/useRejectPVUserMutation.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    onSettled: async (_, __, { currentStatus }) => {
+      const keys = getKeysFromStatus(currentStatus);
+      await queryClient.invalidateQueries({ queryKey: keys });
+      await queryClient.invalidateQueries({ queryKey: [pointsVaultUsers, pointsVaultRejectedUsers] });
+      onSuccess?.();
+    },
+  });
+
+  return { rejectPVUserMutation: mutation };
```
Comment (rohitmalhotra1420) [inline src/queries/hooks/pointsVault/useRejectPVUserMutation.ts:51]: Please follow the same approach written for the useApprovePVUserMutation.ts file

File: src/queries/hooks/pointsVault/useRejectPVUserMutation.ts (hunk 3 above)
Comment (rohitmalhotra1420) [reply src/queries/hooks/pointsVault/useRejectPVUserMutation.ts:23]: The file name and hook name should be same

File: src/queries/hooks/pointsVault/useApprovePVUserMutation.ts (hunk 2 above)
Comment (rohitmalhotra1420) [reply src/queries/hooks/pointsVault/useApprovePVUserMutation.ts:23]: The file name and the hook name should be same.

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]

File: src/modules/pointsVault/components/PointsVaultListContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        variant="h4-bold"
+        color="gray-1000"
+      >
+        Points Vault
+      </Text>
+      <Box
+        display="flex"
+        justifyContent="space-between"
+        alignItems="center"
+        css={[{ marginBottom: 'var(--s6)' }]}
```
[the remark this answers is not in the document]
Comment (kalashshah) [reply src/modules/pointsVault/components/PointsVaultListContainer.tsx:78]: marginBottom is not a supported param yet

File: src/modules/pointsVault/components/PointsVaultList.tsx [hunk 7]
```diff
@@ -1,10 +1,86 @@
...
+  isFetchingNextPage,
+  isError,
+  refetch,
+  fetchNextPage,
+}: PointsVaultListProps) => {
+  const hasMoreData = !isFetchingNextPage && hasNextPage;
+
+  const pointsVaultList = isLoading ? Array(5).fill(0) : data?.pages.flatMap((page: any) => page) || [];
+
+  if (!pointsVaultList.length) {
```
[the remark this answers is not in the document]
Comment (kalashshah) [reply src/modules/pointsVault/components/PointsVaultList.tsx:33]: Will show empty list item even when list is full. Don't need the isSuccess variable

**Review by rohitmalhotra1420 — APPROVED** [review]

**Pull request merged**